dev/tools/localization/localizations_validator.dart

```diff
@@ -103,28 +101,28 @@ void validateLocalizations(
   Map<LocaleInfo, Map<String, String>> localeToResources,
   Map<LocaleInfo, Map<String, dynamic>> localeToAttributes,
 ) {
-  final Map<String, String> canonicalLocalizations = localeToResources[LocaleInfo.fromString('en')];
+  final Map<String, String> canonicalLocalizations = localeToResources[LocaleInfo.fromString('en')]!;
   final Set<String> canonicalKeys = Set<String>.from(canonicalLocalizations.keys);
   final StringBuffer errorMessages = StringBuffer();
   bool explainMissingKeys = false;
   for (final LocaleInfo locale in localeToResources.keys) {
-    final Map<String, String> resources = localeToResources[locale];
+    final Map<String, String>? resources = localeToResources[locale];
```

Comment:
As pointed out before, `resources` should be non-null as all usages of it in the original code assume it wasn't null. In addition it can't be null to start with because the key we are using to look in `localeToResources` is given to us by the `keys` property of `localeToResources`, so there is an entry for it.

Reply:
Yeah sorry for this